week-5/Homework/mandatory/3-project/js/main.js

```diff
@@ -0,0 +1,79 @@
+    JumbotronEl.style.backgroundColor = "#588fbd";
+    donateBykBtn.style.backgroundColor = "#ffa500";
+    volunteerColor.style.backgroundColor = "#000000";
+    volunteerColor.style.color = "#ffffff"
+});
+document.getElementById("orangeBtn").addEventListener("click", function(){// function click the orangeBtn
+    JumbotronEl.style.backgroundColor = "#f0ad4e";
+    donateBykBtn.style.backgroundColor = "#5751fd";
+    volunteerColor.style.backgroundColor = "#31b0d5";
+    volunteerColor.style.color = "#ffffff"
+});
+document.getElementById("greenBtn").addEventListener("click", function(){// function click the greenBtn
+    JumbotronEl.style.backgroundColor = "#87ca8a";
+    donateBykBtn.style.backgroundColor = "#000000";
+    volunteerColor.style.backgroundColor = "#8c9c08";
+});
+
+
+
+////////PART 2
+
+
+/*
+Just below the buttons, there's a form called **Register with us**.
+Continue working in `./js/main.js` to add the following functionality:
+When the submit button is pressed, it should check that all the form fields are valid:
+- The **Your name**, **Email** and **Describe yourself** fields need to be non-empty (Hint: their `value` length has to be greater than zero)
+- For the **Email** field also check if it contains the `@` character
+For all the fields that invalid, it should make their background color `red`.
+IF all the fields are valid, when you click **Submit** it should:
+- Display an alert to thank you for filling out the form
+- Blank out (make empty) all the text fields
+**Important hint:** In your function that handles clicks on the `Submit` button you will need to call `event.preventDefault()` to stop the browser from refreshing the page. To read more on how to do this: https://developer.mozilla.org/en/docs/Web/API/Event/preventDefault
+*/
+
+let allValid = 0;//used this variable to check all inputs valid.
+document.querySelectorAll(".btn-primary")[5].addEventListener("click", function(){// main function click the submit.
+    let emlVal = document.querySelector("#exampleInputEmail1").value;//asigned the value of email input to emlVal.
+    function ValidateEmail(inputText){//function to check the email format.
+var mailformat = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+(inputText.match(mailformat)) ? allValid++ : document.querySelector("#exampleInputEmail1").style.backgroundColor = "red";
+}
+ValidateEmail(emlVal)//call the function validate email.
+let nameVal = document.querySelector("#example-text-input").value;//asigned the value of name input to nameVal.
+(nameVal.length > 0) ? allValid++ : document.querySelector("#example-text-input").style.backgroundColor = "red";
+
+
+let describeVal = document.querySelector("#exampleTextarea").value;
+(describeVal.length > 0) ? allValid++ : document.querySelector("#exampleTextarea").style.backgroundColor = "red";
+
+(allValid === 3) ? alert("thank you for filling out the form") && document.event.preventDefault() : {};//condition to check all inputs are valid.
```

Comment:
you are preventing default only when the form is submitted successfully.
it's more convenient to prevent default in case of errors, you can use `preventDefault()` on both sides of the ternary opereator as 

```javascript
(allValid === 3) ? alert("thank you for filling out the form") && e.preventDefault() :  e.preventDefault();
//condition to check all inputs are valid.
```

Reply:
Great, I got the point